tutorial/examples/cytoscape/usage-phylogeny.py

```diff
@@ -0,0 +1,198 @@
+import math
+import os
+
+import dash_cytoscape
+import dash
+from dash.dependencies import Input, Output
+import dash_html_components as html
+import dash_core_components as dcc
+
+from Bio import Phylo
+
+
+def generate_elements(tree, xlen=30, ylen=30, grabbable=False):
+    def get_col_positions(tree, column_width=80):
+        taxa = tree.get_terminals()
+
+        # Some constants for the drawing calculations
+        max_label_width = max(len(str(taxon)) for taxon in taxa)
+        drawing_width = column_width - max_label_width - 1
+
+        """Create a mapping of each clade to its column position."""
+        depths = tree.depths()
+        # If there are no branch lengths, assume unit branch lengths
+        if not max(depths.values()):
```

Comment:
```suggestion
        if max(depths.values()) == 0:
```
Actually, is it what you mean?
"Use `is not` operator rather than `not ... is`. While both expressions are functionally identical, the former is more readable and preferred.
Reference: https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
Interesting one, I used the code from the Biopython utils and didn't take a close look at this: http://biopython.org/DIST/docs/api/Bio.Phylo._utils-pysrc.html#draw_ascii

Reply:
Honestly I'm not sure about the type of `depths.values()`, since it could be either an empty list, or a singleton list with value of 0. If it's the former, then `if max(depths.values()) is None` is the best idea, as you suggested. However, if it's the former, then `if max(depths.values()) == 0` should work. Although I believe that the behavior should be either one or the other, it's hard for me to say if the `biopython` library has a consistent behavior for depth evaluation (e.g. if the user accidentally sets the branch lengths of every clades in a tree to be 0, then should we force the branch length to be 1, i.e., `depths(unit_branch_lengths=True)`)? I think this is a good issue, and I'll revisit it as soon as I find a greater collection of phylo data so i can thoroughly test the Phylo object.

Reply:
I see! My hesitation was about `depths.values()` being possibly `None`... but then `max(depths.values())` would give a `TypeError` so we'd have to handle the exception (even with the original syntax). So it looks like the case we're handling here really is `max(depths.values()) == 0`. Should we poke the Biopython folks? ;)

Reply:
THat's a good idea! I think I can open up an issue there and share you the link

Reply:
Oops, I didn't see your longer comment meanwhile... Well, `max([0]) == 0` is true and `max([])` gives a `ValueError` so, again, it's about exception handling and/or being more defensive upstream: Do we have some `assert` statements to check the existence and type of `depths.values()` in the Cytoscape code?
Testing popular use cases in the Phylo field is also a good idea!

Reply:
Most of the popular examples contain branch length, which is why I haven't had the chance to extensively test the Phylo object haha

Reply:
https://github.com/biopython/biopython/issues/1910

Reply:
^Just added the link to the issue I just created on the biopython repo

Reply:
Great, @xhlulu! By the way, it's easier if you tag me directly in the issue description, so that I get the initial and subsequent (when people reply) notifications ;)